### PR TITLE
Do not wallog AUX files at replica

### PR DESCRIPTION
**Pull request by knizhnik**
## Problem

Attempt to persist LR stuff at replica cause cannot make new WAL entries during recovery` error.
See https://neondb.slack.com/archives/C07S7RBFVRA/p1729280401283389

## Summary of changes

Do not wallog AUX files at replica.
Related Postgres PRs:

https://github.com/neondatabase/postgres/pull/517
https://github.com/neondatabase/postgres/pull/516
https://github.com/neondatabase/postgres/pull/515
https://github.com/neondatabase/postgres/pull/514

## Checklist before requesting a review

- [ ] I have performed a self-review of my code.
- [ ] If it is a core feature, I have added thorough tests.
- [ ] Do we need to implement analytics? if so did you add the relevant metrics to the dashboard?
- [ ] If this PR requires public announcement, mark it with /release-notes label and add several sentences in this section.

## Checklist before merging

- [ ] Do not forget to reformat commit message to not include the above checklist

### CODE REVIEW

File: test_runner/regress/test_physical_and_logical_replicaiton.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        config_lines=["min_wal_size=32MB", "max_wal_size=64MB"],
+    )
+    s_con = secondary.connect()
+    s_cur = s_con.cursor()
+
+    for pk in range(n_records):
+        p_cur.execute("insert into t (pk) values (%s)", (pk,))
+
+    # Wait at least 15 seconds until LR snapshots are persisted
+    time.sleep(16)
```
Comment (hlinnaka) [inline test_runner/regress/test_physical_and_logical_replicaiton.py:42]: Where does the 15 seconds come from?

File: test_runner/regress/test_physical_and_logical_replicaiton.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    vanilla_pg.start()
+    vanilla_pg.safe_psql("CREATE TABLE t(pk bigint primary key, payload text)")
+    connstr = primary.connstr().replace("'", "''")
+    vanilla_pg.safe_psql(f"create subscription sub1 connection '{connstr}' publication pub1")
+
+    time.sleep(1)
+    secondary = env.endpoints.new_replica_start(
+        origin=primary,
+        endpoint_id="secondary",
+        config_lines=["min_wal_size=32MB", "max_wal_size=64MB"],
```
Comment (hlinnaka) [inline test_runner/regress/test_physical_and_logical_replicaiton.py:33]: Are these min/max_wal_size settings important?

File: test_runner/regress/test_physical_and_logical_replicaiton.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
+from __future__ import annotations
+
+import time
+
+from fixtures.neon_fixtures import NeonEnv, logical_replication_sync
+
+
+def test_physical_and_logical_replication(neon_simple_env: NeonEnv, vanilla_pg):
```
Comment (hlinnaka) [inline test_runner/regress/test_physical_and_logical_replicaiton.py:8]:
```suggestion
def test_physical_and_logical_replication(neon_simple_env: NeonEnv, vanilla_pg):
    """Test read replica of a primary which has a logical replication publication"""
```

**Review by hlinnaka — APPROVED** [review]
The Postgres changes LGTM. The changes in rewriteheap.c are not strictly necessary, as the rewriteheap code cannot be reached in a replica anyway. But it doesn't hurt either.

File: test_runner/regress/test_physical_and_logical_replicaiton.py (hunk 1 above)
Comment (knizhnik) [reply test_runner/regress/test_physical_and_logical_replicaiton.py:43]: 15 seconds is period of dumping LR snapshots.
Actually I am very sorry, but I was not abler to reproduce the reported problem (`Attempt to persist LR stuff at replica cause cannot make new WAL entries during recovery`) with this test. Yes, LR slots are "inherited" by replica. But no subscription can be activated at replica and so no LR workers are started. As a result this slots are inactive and no LR snapshots are saved. 

I still trying to reproduce it...

File: test_runner/regress/test_physical_and_logical_replicaiton.py (hunk 2 above)
Comment (knizhnik) [reply test_runner/regress/test_physical_and_logical_replicaiton.py:34]: Sorry, they have no relation to this PR. This test was mostly copied from another PR (#9458) where this sizes are used to detect that replica is not reusing WAL segments.

File: test_runner/regress/test_physical_and_logical_replicaiton.py (hunk 1 above)
Comment (knizhnik) [reply test_runner/regress/test_physical_and_logical_replicaiton.py]: I have updated the test.
Now it reproduces the problem.

File: test_runner/regress/test_physical_and_logical_replicaiton.py (hunk 2 above)
Comment (knizhnik) [reply test_runner/regress/test_physical_and_logical_replicaiton.py]: > Are these min/max_wal_size settings important?

I removed them.

**Comment by knizhnik** [comment]
@hlinnaka  can you approve please correspondent Postgres PRs?

**Comment by hlinnaka** [comment]
typo on test name: `test_physical_and_logical_replicaiton.py`

**Review by tristan957 — APPROVED** [review]

**Comment by hlinnaka** [comment]
The `test_create_churn_during_restart` test keeps failing. Not always, but often enough

**Pull request merged**